rse-narrative.md

```diff
@@ -0,0 +1,157 @@
+Instead of developing the generic `script_template.py`,
+this chapter could develop `countwords.py` instead.
+(The current version of `countwords.py` will need to be updated to use argparse
+and other libraries/concepts introduced in that chapter.)
+
+### Chapter 5: Git at the command line
+
+The narrative in this chapter currently involves Frances and her colleague Jean Jennings
+writing a history of the ENIAC project.
+Instead, in this chapter the `collate.py` script could be written from scratch,
+tracking changes with Git along the way.
+
+### Chapter 6: Advanced Git
+
+This chapter introduces the concept of branches.
+In the current draft, a violin plot is created on one branch
+and a beeswarm plot on the other.
+The better looking plot is then merged into the master branch.
+This example could be tweaked so that two slightly different versions of 
+`plotcounts.py` are developed instead
+(i.e. two different ways of plotting the word count data).
+
+### Chapter 7: Code style, review and refactor
+
+It's not clear how much of this chapter will ultimately end up in the novice Python book,
+but nonetheless code review will definitely be introduced here.
+The chapter text could be edited such that a draft version of `testfit.py`
+(with deliberate flaws written in)
+is presented and then reviewed/improved.
+
+### Chapter 8: Automating analyses 
+
+This chapter can stay pretty much as is.
+The introductory paragraphs about Zipf's Law can be removed 
+(because they should now appear in the introduction chapter)
+and the workflows developed in the chapter could be extended to also include
+`plotcounts.py` and `testfit.py`.
+
+### Chapter 10: Project structure
+
+Now that we've written all our scripts for the Zipf's Law analysis,
+this chapter should structure it all properly (i.e. with license files and README's etc).
+
+### Chapter 14: Correctness
+
+This chapter introduces testing and could be updated to write tests for the Zipf's Law code.
+For example, `countwords.py` could be tested using an input text file where the count
+for each word is already known.
+
+I think this chapter needs to come before the chapter on continuous integration,
+since it's these tests that Travis CI will implement.
+
+### Chapter 13: Python packaging and Chapter 12: R packaging
+
+In the current version of these chapters,
+a few relevant functions relating to the Zipf's Law analysis 
+(but which haven't been introduced in earlier chapters) are added to a Python module (or R equivalent).
+That module is then packaged so it can be installed by others.
+
+Instead, this chapter could package up all the code developed in the previous chapters
+as a complete Zipf's Law package.
+
+### Chapter 15: Publishing
+
+This lesson could consider (hypothetically) where the results, data and code from the Zipf's Law
+analysis could be published.
+
+
+## Outstanding issues
+
+- Where should "Chapter 9: Working in teams" go in the chapter order?
```

Comment:
Would this chapter maybe go before publishing?

Reply:
I've followed Greg's suggestion and put it after advanced git.